apps/api/src/domain/models/media-item/node-media-management.service.ts

```diff
@@ -0,0 +1,200 @@
+        // TODO check if file exists
+
+        const filepathSplitOnSeparator = filepath.split('.');
+
+        if (filepathSplitOnSeparator.length < 2) {
+            throw new InternalError(
+                `encountered an invalid filepath: ${filepath}. must include an extension`
+            );
+        }
+
+        // TODO what if there are multiple dots in the filename?
+        const filePrefix = filepathSplitOnSeparator[0];
+
+        // TODO this logic needs to be unit tested and separated into a util
+        // also, isn't there a standard lib for doing these manipulations?
+        const filePrefixSplitOnSlashes = filePrefix.split('/');
+
+        const filename = filePrefixSplitOnSlashes[filePrefixSplitOnSlashes.length - 1];
+
+        // TODO use get file extension  helper
+        const extension = filepathSplitOnSeparator[filepathSplitOnSeparator.length - 1];
+
+        if (!isNonEmptyString(extension) || extension.includes('.')) {
+            throw new InternalError(
+                `failed to parse extension for file: ${filepath}. Invalid result: ${extension}`
+            );
+        }
+
+        // does this throw if it doesn't recognize the extension?
+        const expectedMimetypeFromExtension = getExpectedMimeTypeFromExtension(extension);
+
+        const probeResult = await this.mediaProber.probe(filepath);
+
+        if (isNotFound(probeResult)) {
+            throw new Error('todo- return this error');
+        }
+
+        /**
+         * TODO it is critical that we validate the mime type against the extension
+         * before opening up an endpoint to upload binary files
+         */
+        // if (expectedMimetypeFromExtension !== probeResult.mimeType) {
+        //     throw new Error(`todo = return this error?`);
+        // }
+
+        // TODO Execute a command instead
+        const newMediaItem: MediaItem = new MediaItem({
+            // TODO shouldn't this be passed in?
+            title: filePrefix,
+            // TODO remove this
+            type: ResourceType.mediaItem,
+            mimeType: expectedMimetypeFromExtension,
+            // this is totally wrong!
+            published: true,
+            // TODO we need the ID generator- better to just execute the command
+            id: buildDummyUuid(1),
+            // todo remove this!
+            url: 'https://www.ilie.ca/itstruethough',
+        });
+
+        const _errors = newMediaItem.validateInvariants();
+
+        // TODO handle file copy failure
+        // TODO We may want to abstract over file reads and writes with a binary file repository
+        /**
+         * Note that we write the file before persisting to the database because
+         * it is better to have an orphaned file than a database record that
+         * is missing media in case of failure.
+         */
+        copyFileSync(filepath, `${staticAssetsDir}/${filename}.${extension}`);
```

Comment:
I also wonder if we want to copy automatically, or whether we can also run discover "in place" 
(in the static assets directory). We need to be careful around this.